awx/api/views/__init__.py

```diff
@@ -2228,6 +2228,9 @@ class InventorySourceUpdateView(RetrieveAPIView):
 
     def post(self, request, *args, **kwargs):
         obj = self.get_object()
+        serializer = self.get_serializer(instance=obj, data=request.data)
+        if not serializer.is_valid():
```

Comment:
I recommend making use of the `raise_exception=True` parameter instead of constructing your own response.

https://www.django-rest-framework.org/api-guide/serializers/#raising-an-exception-on-invalid-data
https://www.cdrf.co/3.13/rest_framework.generics/CreateAPIView.html#create